packages/blade/src/components/Modal/__tests__/Modal.web.test.tsx

```diff
@@ -0,0 +1,244 @@
+        <ModalFooter>
+          <Button>Save</Button>
+        </ModalFooter>
+      </Modal>,
+    );
+    const container = getByTestId('modal-wrapper');
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should focus on passed initialFocusRef', () => {
+    const Element = (): React.ReactElement => {
+      const ref = React.useRef(null);
+      return (
+        <Modal
+          isOpen={true}
+          onDismiss={() => {}}
+          accessibilityLabel="Test Modal"
+          size="medium"
+          initialFocusRef={ref}
+        >
+          <ModalBody>
+            <Text>Test Content</Text>
+          </ModalBody>
+          <ModalFooter>
+            <Button ref={ref} accessibilityLabel="save">
+              Save
+            </Button>
+          </ModalFooter>
+        </Modal>
+      );
+    };
+
+    const { getByLabelText } = renderWithTheme(<Element />);
+    expect(getByLabelText('save')).toHaveFocus();
+  });
+
+  it('should handle controlled state of Modal', async () => {
+    const user = userEvents.setup();
+    const onDismiss = jest.fn();
+    const Element = (): React.ReactElement => {
+      const [isOpen, setIsOpen] = React.useState(false);
+      return (
+        <>
+          <Modal
+            isOpen={isOpen}
+            onDismiss={() => {
+              console.log('OOONNNNNDISMISS');
+              onDismiss();
+            }}
+            accessibilityLabel="Test Modal"
+            size="medium"
+          >
+            <ModalHeader title="Modal" />
+            <ModalBody>
+              <Text>Test Content</Text>
+            </ModalBody>
+          </Modal>
+          <Button accessibilityLabel="toggle modal" onClick={() => setIsOpen(true)}>
+            Toggle Modal
+          </Button>
+        </>
+      );
+    };
+
+    const { getByRole, queryByRole } = renderWithTheme(<Element />);
+    await user.click(getByRole('button'));
+    expect(queryByRole('dialog')).toBeInTheDocument();
+    await user.keyboard('{Escape}');
+    expect(onDismiss).toHaveBeenCalledTimes(1);
+  });
+
+  it('should dismiss on overlay click', async () => {
+    const user = userEvents.setup();
+    const onDismiss = jest.fn();
+    const Element = (): React.ReactElement => {
+      const [isOpen, setIsOpen] = React.useState(false);
+      return (
+        <>
+          <Modal
+            isOpen={isOpen}
+            onDismiss={onDismiss}
+            accessibilityLabel="Test Modal"
+            size="medium"
+          >
+            <ModalBody>
+              <Text>Test Content</Text>
+            </ModalBody>
+          </Modal>
+          <Button accessibilityLabel="toggle modal" onClick={() => setIsOpen(true)}>
+            Toggle Modal
+          </Button>
+        </>
+      );
+    };
+
+    const { getByRole, getByTestId } = renderWithTheme(<Element />);
+    await user.click(getByRole('button'));
+    await user.click(getByTestId('modal-backdrop'));
+    expect(onDismiss).toHaveBeenCalledTimes(1);
```

Comment:
We can also expect the modal to not be rendering anymore

Reply:
Same as this: https://github.com/razorpay/blade/pull/1251#discussion_r1218955256